apis/v1alpha1/httproute_types.go

```diff
@@ -145,13 +146,23 @@ type HTTPRouteMatch struct {
 	// +optional
 	// +kubebuilder:validation:Enum=Exact;ImplementationSpecific
 	// +kubebuilder:default=Exact
-	HeaderMatchType *string `json:"headerMatchType" protobuf:"bytes,3,opt,name=headerMatchType"`
+	Type *string `json:"type" protobuf:"bytes,3,opt,name=type"`
 	// Headers are the HTTP Headers to match as interpreted via
 	// HeaderMatchType. Multiple headers are ANDed together, meaning, a request
 	// must contain all the headers specified in order to select this route.
 	//
+	// +required
+	Values map[string]string `json:"values" protobuf:"bytes,4,rep,name=values"`
+}
+
+// HTTPRouteMatch defines the predicate used to match requests to a
+// given action.
+type HTTPRouteMatch struct {
+	// +optional
+	Path *HTTPPathMatch `json:"path" protobuf:"bytes,6,opt,name=path"`
+
 	// +optional
-	Headers map[string]string `json:"headers" protobuf:"bytes,4,rep,name=headers"`
+	Header *HTTPHeaderMatch `json:"headers" protobuf:"bytes,7,opt,name=headers"`
```

Comment:
s/headers/header

Reply:
I could go either way on this. To me, `headers` reads better because people colloquially talk about "HTTP Headers", but I could easily be in the minority on this one :)